src/unicode/utf8_everywhere/stdio.cpp

```diff
@@ -80,47 +78,65 @@ inline void set_binary_stdio(FILE* file) THROWS
         throw runtime_exception{ "Could not set STDIO to binary mode." };
 }
 
-void set_console_echo() NOEXCEPT
+#ifndef _WIN32
```

Comment:
Use `HAVE_MSC` as use of vendor PP defines is unsafe, for example:
```
#if defined(_MSC_VER) && !defined(HAVE_CLANG)
    #define HAVE_MSC
#endif
```
See `have.hpp`.